### PR TITLE
Adjusted logs to be consistent across the code base

### LINKED ISSUE #203: Adjust logs?
The operator creates a lot of logs. it seems that every time it runs in a loop it logs this message. The message does not contain any useful information.

```
INFO[0784] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0785] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0785] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0786] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0787] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0787] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0788] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0788] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0789] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0789] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0790] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0791] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0791] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0792] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0793] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0793] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0794] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0794] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0795] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0795] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0796] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0797] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0797] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0798] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0799] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0799] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0800] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0800] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0801] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0802] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0802] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0803] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0803] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0804] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0805] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0805] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0806] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0806] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0807] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0807] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0808] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0809] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0809] Reconciling Jaeger                            name=simple-prod namespace=bar
INFO[0810] Reconciling Jaeger                            name=simple-prod namespace=bar

```

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes #203

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/237)

File: pkg/config/sampling/sampling.go [hunk 1]
```diff
@@ -40,7 +40,10 @@ func (u *Config) Get() *v1.ConfigMap {
 		return nil
 	}
 
-	logrus.WithField("instance", u.jaeger.Name).Debug("Assembling the Sampling configmap")
+	log.WithFields(log.Fields{
```
Comment (pavolloffay) [inline pkg/config/sampling/sampling.go]: I know that you have turned down my opinions but I find the current logging rather ad hoc. I am pretty suer sooner or later somebody will log something without proper fields.

I like static access to logger but in this case I would rather pass it explicitly or use MDC to for consistent logging and avoid repetition every time you access the logger. Another approach is to have a factory method `creteLogger(j v1alpha1.Jaeger, tuples...) logger` in utils package.

Comment (jpkrohling) [reply pkg/config/sampling/sampling.go]: > somebody will log something without proper fields

This happens already, as this PR shows: quite a few places had to be fixed to include the instance and/or namespace.

I'm not against using a base logger for the request-scoped code, but I don't quite like the idea of changing method signatures to include a logger which can be derived from the other parameters (`log.WithFields("name": jaeger.Name, "namespace":jaeger.Namespace)`)

A helper like you proposed would certainly be something to consider, though.

Comment (jpkrohling) [reply pkg/config/sampling/sampling.go]: I added a new commit in the direction of your suggestion. How does it look now?

Comment (pavolloffay) [reply pkg/config/sampling/sampling.go]: it looks much better :+1:

**Comment by pavolloffay** [comment]
Does this resolve #203? My concert in #203 is that the log just says `Reconciling...`. and if the loop is invoked often like in OCP #231 it produces a high volume of logs with no valuable information. 

I would expect it log something if there is a difference in desired state.

But this PR looks good it's definitelly an improvement.

**Comment by jpkrohling** [comment]
At the debug level, I think the `Reconciling...` message is useful, to know that the loop has started. The fact that it shows up hundred of times should be considered a bug, which is handled by #231. In any case, it has now an `execution` field, so that we can correlate all log events for a given reconciliation loop.

I see #203 as a general "logging" issue, which is what this PR is solving.

> I would expect it log something if there is a difference in desired state.

Hopefully done as part of #231 :-)

**Pull request merged**